### PR TITLE
Move data objects and collections

### PR DESCRIPTION
**Pull request by chStaiger**
New function in IrodsPath to move/rename data objects and collections.

### CODE REVIEW

File: docker/irods_client/tests/test_move.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+@mark.parametrize("item_name", ["collection", "dataobject"])
+def test_move(session, item_name, request):
+    item = request.getfixturevalue(item_name)
+    old_path = IrodsPath(session, item.path)
+    new_path = IrodsPath(session, '~', 'moved')
+
+    old_path.move(new_path)
+    assert new_path.exists()
+
```
Comment (qubixes) [inline docker/irods_client/tests/test_move.py:13]: We should assert `not old_path.exists()`.

File: docker/irods_client/tests/test_move.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    new_path.move(old_path)
+    assert old_path.exists()
```
Comment (qubixes) [inline docker/irods_client/tests/test_move.py:15]: Same here.

**Review by qubixes — CHANGES_REQUESTED** [review]
Very nice feature! CLI version incoming?!

File: ibridges/path.py [hunk 3]
```diff
@@ -136,9 +136,43 @@ def create_collection(session,
             raise PermissionError(
                 "While creating collection '{coll_path}': iRODS server forbids action.") from exc
 
-    def rename(self):
-        """Rename the collection or data object."""
-        raise NotImplementedError("Rename not implemented yet.")
+    def move(self, new_name: Union[str, IrodsPath]):
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/path.py:139]: You mean `rename`, do you?

[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/path.py:139]: Done! And I also renamed the function to `rename`.

[the remark this answers is not in the document]
Comment (qubixes) [reply ibridges/path.py:139]: Ah yes!

**Pull request merged**